### PR TITLE
docs: describe BelongsTo relation

### LINKED ISSUE #1361: [Relation] Support BelongsTo relation
## Description / Steps to reproduce / Feature proposal
Add support for `BelongsTo` relation between two models. Stretch goal for `DP3`.

### Acceptance Criteria
- [x] Update [factory method](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/repositories/relation.factory.ts) to handle this relation
- [x] create interface for CRUD methods available for the relation.  See [this](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/repositories/relation.repository.ts) as an example.
    - [x] create default class which implements the interface
- [x] Implement the `belongsTo` decorator
  - [x] model decorator: infers the metadata for the relation
  - [x] repository decorator: return the constrained target crud repository (possible name, `belongsToRepository`)
- [x] Address dual usage of `hasMany` and `belongsTo` for a two way relation definition
- [ ] tests / docs / blog post

### PR DESCRIPTION
**Pull request by bajtos**
Close #1361 🎉 

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: docs/site/BelongsTo-relation.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+permalink: /doc/en/lb4/BelongsTo-relation.html
+---
+
+## Overview
+
+A `belongsTo` relation denotes a many-to-one connection of a model to another
+model through referential integrity. The referential integrity is enforced by a
+foreign key constraint on the source model which usually references a primary
+key on the target model. This relation indicates that each instance of the
+declaring or source model belongs to exactly one instance of the target model.
```
Comment (jannyHou) [inline docs/site/BelongsTo-relation.md]: hmm why it's `declaring or source` here? I understand the source model, but wondering what does "declaring model" mean..

File: docs/site/BelongsTo-relation.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+To add a `belongsTo` relation to your LoopBack application and expose its
+related routes, you need to perform the following steps:
+
+1. Add a property to your source model to define the foreign key.
+2. Modify the source model repository class to provide an accessor function for
+   obtaining the target model instance.
+3. Call the accessor function in your controller methods.
+
+## Defining a hasMany Relation
```
Comment (jannyHou) [inline docs/site/BelongsTo-relation.md]: typo? ## Defining a belongsTo Relation

File: docs/site/BelongsTo-relation.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  ): Promise<Customer> {
+    return await this.orderRepository.customer(orderId);
+  }
+}
+```
+
+In LoopBack 3, the REST APIs for relations were exposed using static methods
+with the name following the pattern `__{methodName}__{relationName}__` (e.g.
+`Order.__get__customer`). While we recommend to create a new controller for each
+hasMany relation in LoopBack 4, we also think it's best to use the main CRUD
```
Comment (jannyHou) [inline docs/site/BelongsTo-relation.md]: hasMany --> belongsTo?

File: docs/site/BelongsTo-relation.md (hunk 1 above)
Comment (bajtos) [reply docs/site/BelongsTo-relation.md]: I have copied this from HasMany docs.

My understanding: "declaring model" and "source models" are referring to the same thing - the model class where the relation is declared.

I am happy to improve this sentence, what do you propose?

File: docs/site/BelongsTo-relation.md (hunk 2 above)
Comment (bajtos) [reply docs/site/BelongsTo-relation.md]: Good catch!

File: docs/site/BelongsTo-relation.md (hunk 3 above)
Comment (bajtos) [reply docs/site/BelongsTo-relation.md]: LOL, I should run a search for "hasMany" after creating this page as a copy of HasMany-relation.md.

Thank you for catching this.

Comment (bajtos) [reply docs/site/BelongsTo-relation.md]: Actually, "hasMany" is correctly used here. The full sentence:

> While we recommend to create a new controller for each hasMany relation in LoopBack 4, we also think it's best to use the main CRUD controller as the place where to explose `belongsTo` API.

Comment (jannyHou) [reply docs/site/BelongsTo-relation.md]: Ah I see :-p

**Pull request merged**